lib/delegate/index.js

```diff
@@ -180,14 +184,16 @@ const createSubOperationDocument = function (component, targetRootField, args, s
     }
   }
 
-  
-  // prune fields in the delegated selection set that are not defined
-  // in the schema we are delegating to
-  visitFunctions[Kind.FIELD] = function () {
-    const fieldDef = typeInfo.getFieldDef();
-    // if the field isn't defined in delegated schema, return null to remove the field from the delegated document
-    if (!fieldDef) {
-      return null;
+  // prune selection set fields that are not defined in the target schema
+  visitFunctions[Kind.FIELD] = function (node) {
+    const parentType = typeInfo.getParentType();
+    if (isObjectType(parentType) || isInterfaceType(parentType)) {
```

Comment:
Will this check be isAbstractType() instead of just isInterfaceType() to cover both Union and Interfaces?

Reply:
nice catch - I'll change this